### ISSUE TITLE
refactor documentation build system to use Makefile, as before cmake

### ISSUE THREAD

**Issue by prjemian** [nexusformat/definitions#417]
The current documentation build system, cmake, has a problem as noted in #407.

Re-develop the documentation build system using the infrastructure we had before cmake, using Makefiles and Python scripts.  This should run on Linux, MacOSX, and Windows.

Extra points for an out-of-source-tree build.

**Comment by prjemian** [comment]
Here is the Make for Windows recommended for EPICS

* Download and install Make (http://gnuwin32.sourceforge.net/packages/make.htm)

I used this to build NeXus documentation on Windows previously

source: http://www.aps.anl.gov/epics/tech-talk/2010/msg00595.php

**Comment by prjemian** [comment]
commit 70e5eb seems to be a good reference before cmake changes were started

**Comment by prjemian** [comment]
better: https://github.com/nexusformat/definitions/tree/ffccfce66ab1e22b4b4ee6dc0afae01c9364873a

**Comment by prjemian** [comment]
With commit c095c3c, the documentation builds from the command line on both linux (linuxmint v17.2) and Windows (Win 8.1 x86_64, using cmd.exe, not powershell).

On Windows, if the following error is output to the console after typing ```make html``` in the <definitions>/manual directory:

```
'python' is not recognized as an internal or external command,
operable program or batch file.
```
it may be necessary to add Python to the PATH environment variable, such as one of these lines:

```
set PATH=C:\Python27;%PATH%
set PATH=D:\Apps\Anaconda;%PATH%
```

Try to avoid installing Python on a path that has embedded spaces.  Long series of tedious problems with that.

**Comment by prjemian** [comment]
Need to replace @NXDL_RELEASE@ with current NXDL version number in manual/source/conf.py

**Comment by prjemian** [comment]
Need to remove cmake configuration files

**Comment by prjemian** [comment]
Look at this commit for non-CMake configs
https://github.com/nexusformat/definitions/blob/e6d701d90e9569914607ba0b692ab9f856ab5e04/applications/NXdirecttof.nxdl.xml

**Comment by prjemian** [comment]
testing latest on office PC (RHEL6), problems:

```
jemian@gov .../classes/base_classes $ make
cp ../../../../base_classes/NXattenuator.nxdl.xml   ./NXattenuator.nxdl.xml
python ../../../../utils/nxdl2rst.py NXattenuator.nxdl.xml > NXattenuator.rst
Traceback (most recent call last):
  File "../../../../utils/nxdl2rst.py", line 395, in <module>
    main()
  File "../../../../utils/nxdl2rst.py", line 379, in main
    print_rst_from_nxdl(nxdl_file)
  File "../../../../utils/nxdl2rst.py", line 308, in print_rst_from_nxdl
    printDoc(INDENTATION_UNIT, ns, root, required=True)
  File "../../../../utils/nxdl2rst.py", line 159, in printDoc
    raise Exception( 'No documentation for: ' + node.get('name') )
Exception: No documentation for: NXattenuator
make: *** [NXattenuator.rst] Error 1
```

need to investigate and resolve before making a PR

**Comment by prjemian** [comment]
resolved build problem with fa73c77
tested on Linux and Windows (assume MacOSX will work, do not have host to test)
ready for PR
